src/Bicep.Core/TypeSystem/ResourceTypeRegistrar.cs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using Bicep.Core.Resources;
+using Bicep.Core.TypeSystem.Azrm;
+
+namespace Bicep.Core.TypeSystem
+{
+    public class ResourceTypeRegistrar
+    {
+        public static ResourceTypeRegistrar Instance { get; } = new ResourceTypeRegistrar();
```

Comment:
>public static ResourceTypeRegistrar Instance { get; } = new ResourceTypeRegistrar(); [](start = 8, length = 84)

Let's not do a static singleton. It'll prevent us from being able to inject a mock for testing.

Reply:
And it's also hard to get rid of later.

---
In reply to: [493893097](https://github.com/Azure/bicep/pull/553#discussion_r493893097) [](ancestors = 493893097)